### ISSUE TITLE
<Fields /> component field-level validation

### ISSUE THREAD

**Issue by pinguinjkeke** [erikras/redux-form#3492]
### Are you submitting a **bug report** or a **feature request**?
Feature request.

### What is the current behavior?
I have dynamically changing field and I'm using multiple fields ```<Fields />``` component as it perfectly fits my requirements instead of validation.
I'm trying to pass validation array for a current active field or object, where the keys are fields names and values are validation arrays respectively, but nothing happens and redux store is telling me nothing about wrong values.

### What is the expected behavior?
Using of field-level validation in multiple fields ```<Fields />``` component by passing an array of validation functions for current active field or object representing validation for every field.

### What's your environment?
I'm using latest version of redux-form (7.1.0 currently).

**Comment by Azein** [comment]
I have no clue how to validate <Fields /> on field-level currently, so I would be very happy if there is a future for enhancements like this one.

**Comment by DavidHooper** [comment]
@Azein A field level validation example can be found here https://redux-form.com/7.1.2/examples/fieldlevelvalidation/

**Comment by Azein** [comment]
@DavidHooper 
  You are right, but we are talking about <Fields />, not <Field />. And I haven't seen an example of <Fields /> field-level validation yet. For now I validate <Fields /> stuff in general validate function for the form, thus combining field-level (for other components) and form-level approaches.

**Comment by indacloud** [comment]
I had the same issue and took a look to the code... It's nicely written and easy to understand :+1: .

I came up with something working for me but I'd like some feedback and/or ideas to improve it. Also some help for `flow` typing is welcome.

https://github.com/indacloud/redux-form/pull/1

Thanks !

**Comment by iddan** [comment]
Has anyone reviewed this?

**Comment by tukkajukka** [comment]
@indacloud, looks like your PR points to the master branch of your own fork, and not to the master branch of this repo. Might be the reason why it has not been noticed here. :-)

**Comment by malyfko** [comment]
As a workaround I found a working solution (for sync validation): in (this)[https://redux-form.com/6.6.3/examples/syncvalidation/] example author describes how to write validation func and pass it to redux-form with initialization. It doesn't override the `syncErrors` object, but rather 'adds-up'. So what I did is I wrote and passed to redux-form a `validate` func for each field in `Fields`. In my example I have a date range as a single component with 2 fields - `start` and `end`. So the func looks like this:
```
const validate = (values) => {
  const errors = {};
  if (!values.start) {
    errors.start = 'Required field';
  }

  if (!values.end) {
    errors.end = 'Required field';
  }

  return errors;
};
```
Hope I helped 🙂
